src/metaswitch/clearwater/cluster_manager/plugin_utils.py

```diff
@@ -166,15 +165,14 @@ def start_cassandra():
 
     # Wait until we can connect on port 9160 - i.e. Cassandra is running.
     while True:
-        try:
-            if cassandra_not_monitored:
-                # The monit command can fail because monit is still processing
-                # the unmonitor command from before (even though it has
-                # finished unmonitoring cassandra)
-                rc = run_command("monit monitor -g cassandra")
-                cassandra_not_monitored = (rc != 0)
-            elif can_contact_cassandra():
-                break
+        if cassandra_not_monitored:
```

Comment:
If cassandra is unmonitored, I think this will run `monit monitor -g cassandra` in a tight loop (we'll never hit the sleep).

Either don't have the time.sleep in the else clause, or:

```
while cassandra_not_monitored or not (can_contact_cassandra()):
    if cassandra_not_monitored:
        rc = run_command("monit monitor -g cassandra")
        cassandra_not_monitored = (rc != 0)
   time.sleep(1)
```